plugins/plugin_loader.cpp

```diff
@@ -52,34 +64,42 @@ void PluginLoader::LoadPlugins(const std::filesystem::path& plugins_dir_path)
             continue;
         }
 
-        NativeLibraryHandle handle = m_library_loader->Load(file_path);
-        if (handle == nullptr)
+        absl::StatusOr<NativeLibraryHandle> handle_status = m_library_loader->Load(file_path);
+        if (!handle_status.ok())
         {
-            std::cout << "PluginLoader: Failed to load library: " << file_path
-                      << " Error: " << m_library_loader->GetLastErrorString() << std::endl;
+            append_error_message(std::string(handle_status.status().message()));
             continue;
         }
+        NativeLibraryHandle handle = handle_status.value();
 
         LibraryHandleUniquePtr library_handle_ptr(handle,
                                                   NativeLibraryHandleDeleter(
                                                   m_library_loader.get()));
 
+        absl::StatusOr<void*>
+        create_func_symbol_status = m_library_loader->GetSymbol(handle, "CreateDivePluginInstance");
```

Comment:
nit: abseil recommends treating StatusOr like a smart pointer, so their recommendation for naming is to just call it the thing without a suffix like `_or` or, in this case, `_status`. https://abseil.io/tips/181#avoiding-the-_or-suffix
```suggestion
        create_func_symbol = m_library_loader->GetSymbol(handle, "CreateDivePluginInstance");
```

Reply:
fixed